src/java/org/apache/cassandra/cql3/QueryProcessor.java

```diff
@@ -391,6 +397,61 @@ public static UntypedResultSet executeInternal(String query, Object... values)
             return null;
     }
 
+    public static Future<UntypedResultSet> execute(InetAddressAndPort address, String query, Object... values)
```

Comment:
the reason for async is so repair can submit and deal with when complete; rather than block waiting.